tests/architecture/Test.yaml

```diff
@@ -0,0 +1,18 @@
+rules:
+  testAssertionsImplementAssertionInterface:
+    - classes:
+        - havePath: Rule/Assertion/*
+    - excluding:
+        - haveClassName: PhpAT\Rule\Assertion\*\MustNot*
+        - havePath: Rule/Assertion/MatchResult.php
+        - havePath: Rule/Assertion/AbstractAssertion.php
+    - assert: mustExtend
+    - classes:
+        - haveClassName: PhpAT\Rule\Assertion\AbstractAssertion
+
+  testExtractorsDependOnRuleBuilder:
+    - classes:
+        - implementInterface: PhpAT\Test\TestExtractor
+    - assert: mustDependOn
+    - classes:
```

Comment:
I think it'd be useful to have a different name for the 2 `classes` key. Otherwise, it could be confusing and easy to mistake in a larger config even if it works implementation-wise.

What do you think about `classes` and `assertClasses`? Or we could nest the assert `classes` under `assert`? But even I'd prefer to have a different key to reduce confusion.

Reply:
As for now, I would stick to the naming conventions on the PHP classes used for testing, as from my point of view the YAML should follow the same naming as PHP classes, to avoid confusion between the different ways of defining the tests. And also making it easier to switch from one type of config to another one. And also order matters in this type of config as in PHP config is not the same to put the classes before or after the assertion. So keeping this in mind I would say that it is pretty safe to have both `classes` key, as the order is what matters.